Overview page - Intro to quantum computing (#794)
MicrosoftDocs/quantum-docs-pr: articles/overview/overview.md

---
title: Introduction to quantum computing
description: Learn what quantum computing is, what quantum computers can do, and how you can learn quantum computing.
author: bradben
ms.author:  bradben
ms.date: 04/22/2020
ms.topic: overview
uid: microsoft.quantum.overview.introduction
---

# Introduction to quantum computing 

> There are some problems so difficult, so incredibly vast, that even if every supercomputer in the world worked on the problem, it would still take longer than the lifetime of the universe to solve.

Reviewer: This feels a bit over the top; I'd suggest toning it down a bit, especially for an opening paragraph.

Reviewer 2: Agree. Tone it down. Something like: for certain tasks, our most powerful computers run into limitations. 
BTW, would you like to use my cartoons in this page? [Kitty's Hackaday comics for hobbyists](https://hackaday.io/project/168554/logs?sort=newest&page=2)

> Quantum computers hold the promise to solve some of our planet's biggest challenges - in environment, agriculture, health, energy, climate, materials science, and problems not yet even imagined. The impact of quantum computers will be far-reaching and have as great an impact as the creation of the transistor in 1947, which paved the way for today’s digital economy.

Reviewer: Similarly here, I would suggest against making concrete claims comparing the long-reaching historical impact of quantum computing to specific other technologies; especially given that quantum computers still make use of classical computers, the transistor is a tough bar to match.

## What is quantum computing?

> In modern usage, the word quantum means the _smallest possible discrete unit of any physical property_, usually referring to atomic or sub-atomic particles. Quantum computers use actual quantum particles – atoms or ions – as processing units, and are large, complex, and expensive devices. 

Reviewer: Superconducting qubits and topological qubits tend are more macroscopic than individual quanta. Even atoms and ions are not the individual quanta, but are comprised of many subatomic particles, each of which themselves are composed of quarks, gluons, etc.

> In modern usage, the word quantum means the _smallest possible discrete unit of any physical property_, usually referring to atomic or sub-atomic particles. Quantum computers use actual quantum particles – atoms or ions – as processing units, and are large, complex, and expensive devices. 

Reviewer 2: ```suggestion
In modern usage, the word quantum means the _smallest possible discrete unit of any physical property_, usually referring to properties of atomic or sub-atomic particles. Quantum computers use actual quantum particles, artificial atoms or collective properties of quantum particles, as processing units, and are large, complex, and expensive devices. 
```

> Harnessing the unique behavior of quantum physics and applying it to computational computing, quantum computers introduce new concepts to traditional programming methods, making use of quantum physics behavior such as superposition, entanglement, and quantum interference. 

Reviewer 2: ```suggestion
Harnessing the unique behavior of quantum physics and applying it to computing, quantum computers introduce new concepts to traditional programming methods, making use of quantum physics behaviors such as superposition, entanglement, and quantum interference. 
```

## What can a quantum computer do?

A quantum computer isn’t a supercomputer that can do everything faster, it just does some select things extremely well. 

### Quantum simulation

> Since quantum computers use quantum particles as their processors, they are well-suited for modeling other quantum systems. Photosynthesis, superconductors, and complex molecules are examples of quantum systems that can be simulated using quantum programs.

Reviewer: Transistors take advantage of quantum effects as well, but aren't as well-suited. It's more that the way qubits behave requires a quantum description...

> Since quantum computers use quantum particles as their processors, they are well-suited for modeling other quantum systems. Photosynthesis, superconductors, and complex molecules are examples of quantum systems that can be simulated using quantum programs.

Reviewer 2: ```suggestion
Since quantum computers use quantum phenomena in computation, they are well-suited for modeling other quantum systems. Photosynthesis, superconductivity, and complex molecular formation are examples of quantum mechanisms that can be simulated using quantum programs.
```

### Cryptography and Shor’s algorithm

> In 1994, Peter Shor showed that a scalable quantum computer can break every widely used encryption technique. Classical cryptography relies on the intractability of operations on large numbers, such as factorization of large numbers into two prime numbers, and quantum computers excel at factorization. 

Reviewer: Links to our samples could work well here.

### Search and Grover’s algorithm

> In 1996, Lov Grover developed a quantum algorithm that dramatically sped up the solution to unstructured searches faster than any classical algorithm can do. This increased speed becomes significant in lists that contain millions or billions of items.

Reviewer: I'd again suggest linking to samples; this also needs a bit of clarification to avoid inadvertently overpromising.

### Optimization

> Optimization is the task of balancing a system for maximum efficiency and minimum cost. A quantum computer can speed up optimization algorithms and find solutions that otherwise were not possible. In addition to optimizing traffic flow to reduce congestion, there is airplane gate assignment, package delivery, job scheduling and more. With breakthroughs in materials science, there will be new forms of energy, batteries with greater capacity, lighter and stronger materials.

Reviewer 2: ```suggestion
Optimization is the task of balancing a system for maximum efficiency and minimum cost. A quantum computer can speed up optimization algorithms and find solutions that otherwise were not possible. In addition to optimizing traffic flow to reduce congestion, there is airplane gate assignment, package delivery, job scheduling and more. With breakthroughs in materials science, there will be new forms of energy, batteries with greater capacity, and lighter and stronger materials.
```

> ### Machine Learning

Reviewer: I'd suggest making the capitalization here consistent with everything else.
```suggestion
### Machine learning
```

> A majority of numerical calculations in classical computing consists of solving linear systems of equations. This is especially true in the field of machine learning where most of the computation cost goes into calculating the inverse of huge matrices.

Reviewer: Is it true that most classical computing is spent on linear equations? Do we have a good reference for that?

> Fortunately, there is a quantum algorithm that allows us to approximate a solution exponentially faster than a classical computer. Reducing computation times in these areas will help address the energy crisis, climate change, food scarcity, and personal and precise medical diagnosis.

Reviewer: I would not make the strong claim of an exponential speedup here. +@alexeib2 for QML impact.

Reviewer 3: What about this https://arxiv.org/abs/0811.3171?

> ### Quantum-inspired computing

Reviewer: +@anraman and @frtibble for QIO impact.

Reviewer 2: Are we talking about QIO or quantum-inspired computing (with the latter being broader?) And earlier we've also talked about Optimization. Shall we combine the two?

Quantum-inspired algorithms use quantum principles for increased speed and accuracy, but are implemented on classical computer systems.

> For example, using algorithms developed by Microsoft’s quantum computing team, researchers at Case Western Reserve University have produced MRI scans that are up to three times faster than prior state-of-the-art approaches, as well as scans that are almost 30 percent more precise in measuring a key identifier of disease.

Reviewer: It'd be good to link to news articles and such here.

## Current advancements

> Recently, Microsoft’s Quantum Lab located at the University of Sydney made breakthroughs in qubit control technology that allow them to scale beyond the physical limitations of current conventional systems. 

Reviewer: This feels like an odd pair of advancements to select out of everything Microsoft has done in this area; if we keep this section, I'd suggest workshopping with @cpalmer2020 on this one to find some other appropriate achievements to pull in.

Reviewer 2: Also, "physical limitations of current conventional systems" for what, temperature, precision of qubit manipulation or? Some explanation of what our control system has improved will be helpful.

> In 2019, Microsoft and IonQ, a quantum computer manufacturer, announced a partnership to make their laser-based quantum computers available to developers and researchers via the cloud. Users will be able to use Microsoft’s QDK and Q# to write quantum programs and run them on IonQ hardware. 

Reviewer: IonQ is one of several partners, and I'm not sure I'd summarize ion traps as being "laser-based." It would be good to get @cpalmer2020's feedback on this.

Author: Perhaps a more accurate wording would be "laser-controlled"? Or at this point, not worry about introducing those details and just refer to "... to make their quantums computers available..."

Reviewer 2: We should to mention all the partners.

Reviewer 4: I agree, probably easiest to avoid the specifics of the implementation methods at this point. But mentioning all the partners also lets us highlight that our partnerships span both of the most promising system types: trapped-ion qubits (IonQ and Honeywell) and superconducting qubits (QCI).

## Q# and the Microsoft Quantum Development Kit (QDK)

> Q# is Microsoft’s open-source programming language for developing and running quantum algorithms. It is part of the QDK, a full-featured development kit for Q# that you can use with common tools and languages to develop quantum applications that you can run in various environments.

Reviewer: I'd suggest being specific here and highlighting Jupyter, Python, C#, VS 2019, VS Code...

Author: agree

---
title: Introduction to quantum computing
description: Learn what quantum computing is, what quantum computers can do, and how you can learn quantum computing.
author: bradben
ms.author:  bradben

> ms.date: 04/22/2020

Reviewer 5: ```suggestion
ms.date: 05/05/2020
```

For some of these problems, even our most powerful computers run into problems. While quantum technology is just beginning to impact the computing world, it could be far-reaching and change the way we think about computing.

## What is quantum computing?

In modern usage, the word quantum means the smallest possible discrete unit of any physical property, usually referring to properties of atomic or sub-atomic particles. Quantum computers use actual quantum particles, artificial atoms, or collective properties of quantum particles as processing units, and are large, complex, and expensive devices.

Harnessing the unique behavior of quantum physics and applying it to computing, quantum computers introduce new concepts to traditional programming methods, making use of quantum physics behaviors such as superposition, entanglement, and quantum interference.

## What can a quantum computer do?

> A quantum computer isn't a supercomputer that can do everything faster: it just does some select things exceptionally well.

Reviewer 5: ```suggestion
A quantum computer isn't a supercomputer that can do everything faster; it just does some select things exceptionally well.
```

Optimization is the task of balancing a system for maximum efficiency and minimum cost and is a crucial issue for industry and science. Quantum-inspired optimization algorithms running on today's classical computers can find solutions that up to now have not been possible. In addition to optimizing traffic flow to reduce congestion, there is airplane gate assignment, package delivery, job scheduling and more. With breakthroughs in materials science, there will be new forms of energy, batteries with larger capacity, and lighter and more durable materials.

> [!NOTE]
> Read more about how Microsoft quantum-inspired computing is being used in [materials science](https://cloudblogs.microsoft.com/quantum/2020/01/21/oti-lumionics-accelerating-materials-design-microsoft-azure-quantum/), [risk management](https://cloudblogs.microsoft.com/quantum/2019/05/22/microsoft-quantum-collaborates-with-willis-towers-watson-to-transform-risk-management-solutions/), and [medicine](https://blogs.microsoft.com/blog/2018/05/18/microsoft-quantum-helps-case-western-reserve-university-advance-mri-research/).

> ### Machine learning

Reviewer 5: ```suggestion
### Quantum machine learning
```

The Quantum Development Kit comes with the Quantum Machine Learning Library that gives you the ability to run hybrid quantum/classical machine learning experiments. The library includes samples and tutorials, and provides the necessary tools to implement a new hybrid quantum-classical algorithm, the circuit-centric quantum classifier, to solve supervised classification problems.

## Q# and the Microsoft Quantum Development Kit (QDK)

Q# is Microsoft's open-source programming language for developing and running quantum algorithms. It is part of the QDK, a full-featured development kit for Q# that you can use with standard tools and languages to develop quantum applications that you can run in various environments, including the built-in full-state quantum simulator.

There are extensions for Visual Studio and VS Code, and packages for use with Python and Jupyter Notebooks.

The QDK includes a standard library along with customized chemistry, machine learning, and numerics libraries.

> The documentation includes a Q# language guide, tutorials, and sample code to get you started quickly, and rich articles to help you dive deeper into quantum computing concepts.  

Reviewer: Links here would be helpful as well, I'd suggest.

## Microsoft quantum hardware partners

Microsoft is partnering with quantum hardware companies to provide developers with cloud access to quantum computers. Leveraging the [Azure Quantum](https://azure.microsoft.com/services/quantum/) platform and the Q# language, developers will be able to explore quantum algorithms and run their quantum programs on different types of quantum hardware.

> [IonQ](https://ionq.com/news/november-4-2019-microsoft-partnership) and [Honeywell](https://www.honeywell.com/en-us/newsroom/news/2019/11/the-future-of-quantum-computing) both use **trapped ion** based processors, utilizing individual ions trapped in an electronic field, whereas [QCI](https://quantumcircuits.com/news-and-publications/quantum-circuits-partners-with-microsoft-on-azure-quantum) uses superconducting metal circuits.

Reviewer: ```suggestion
[IonQ](https://ionq.com/news/november-4-2019-microsoft-partnership) and [Honeywell](https://www.honeywell.com/en-us/newsroom/news/2019/11/the-future-of-quantum-computing) both use **trapped ion** based processors, utilizing individual ions trapped in an electronic field, whereas [QCI](https://quantumcircuits.com/news-and-publications/quantum-circuits-partners-with-microsoft-on-azure-quantum) uses superconducting circuits.
```

Harnessing the unique behavior of quantum physics and applying it to computing, quantum computers introduce new concepts to traditional programming methods, making use of quantum physics behaviors such as superposition, entanglement, and quantum interference.

## What can a quantum computer do?

A quantum computer isn't a supercomputer that can do everything faster: it just does some select things exceptionally well.

### Quantum simulation

Since quantum computers use quantum phenomena in computation, they are well-suited for modeling other quantum systems. Photosynthesis, superconductivity, and complex molecular formations are examples of quantum mechanisms that quantum programs can simulate.

### Cryptography and Shor’s algorithm

In 1994, Peter Shor showed that a scalable quantum computer could break every widely used encryption technique. Classical cryptography relies on the intractability of operations on large numbers, such as factorization of large numbers into two prime numbers, and quantum computers excel at factorization.

### Search and Grover’s algorithm

> In 1996, Lov Grover developed a quantum algorithm that dramatically sped up the solution to unstructured data searches, running the search in fewer steps than any classical algorithm could. This increased speed becomes significant in lists that contain millions or billions of items.

Reviewer 3: ```suggestion
In 1996, Lov Grover developed a quantum algorithm that dramatically sped up the solution to unstructured data searches, running the search in fewer steps than any classical algorithm could.
```
I would remove that sentence since the use of quantum computers for searching in databases is still very speculative and not very likely. The main application of Grover's algorithm is to search for solutions for NP problems and not searching in physical databases. Here a brief discussion from Niels and Chuang's book.

![image](https://user-images.githubusercontent.com/48300381/80979985-5854a600-8e28-11ea-971d-0094f61284dc.png)

Author: Done, thanks.

> [!NOTE] 
> Ready to try it out? Jump to the Q# tutorial [Implement Grover's search algorithm in Q#](xref:microsoft.quantum.quickstarts.search).

### Machine Learning

> A majority of numerical calculations in classical computing consists of solving linear systems of equations. This is especially true in the field of machine learning where most of the computation cost goes into calculating the inverse of huge matrices.

Reviewer 3: ```suggestion
A great number of numerical calculations in classical computing consists of solving linear systems of equations. This is especially true in the field of machine learning where most of the computation cost goes into calculating the inverse of huge matrices.
```

## What is quantum computing?

In modern usage, the word quantum means the smallest possible discrete unit of any physical property, usually referring to properties of atomic or sub-atomic particles. Quantum computers use actual quantum particles, artificial atoms, or collective properties of quantum particles as processing units, and are large, complex, and expensive devices.

Harnessing the unique behavior of quantum physics and applying it to computing, quantum computers introduce new concepts to traditional programming methods, making use of quantum physics behaviors such as superposition, entanglement, and quantum interference.

## What can a quantum computer do?

A quantum computer isn't a supercomputer that can do everything faster, but there are a few areas where quantum computers do exceptionally well.

- [Quantum simulation](xref:microsoft.quantum.overview.introduction#quantum-simulation)
- [Cryptography](xref:microsoft.quantum.overview.introduction#cryptography-and-shors-algorithm)
- [Search](xref:microsoft.quantum.overview.introduction#search-and-grovers-algorithm)

> - [Optimization](xref:microsoft.quantum.overview.introduction#quantum-inspired-computing-and-optimization)

Reviewer: It sounds from the xref like this is more about QIO than what a quantum computer can do, such that this line item may be better under a different header.

Author: This is just an in-topic link to the Optimization subsection.

- [Machine learning](xref:microsoft.quantum.overview.introduction#quantum-machine-learning)

## Quantum simulation

Since quantum computers use quantum phenomena in computation, they are well-suited for modeling other quantum systems. Photosynthesis, superconductivity, and complex molecular formations are examples of quantum mechanisms that quantum programs can simulate.

## Cryptography and Shor’s algorithm

In 1994, Peter Shor showed that a scalable quantum computer could break widely used encryption techniques such as the RSA algorithm. Classical cryptography relies on the intractability of problems such as integer factorization or discrete logarithms, many of which can be solved more efficiently using quantum computers.

## Search and Grover’s algorithm

In 1996, Lov Grover developed a quantum algorithm that dramatically sped up the solution to unstructured data searches, running the search in fewer steps than any classical algorithm could.

> [!NOTE]
> Ready to try it out? Jump to the Q# tutorial [Implement Grover's search algorithm in Q#](xref:microsoft.quantum.quickstarts.search).

> ## Quantum-inspired computing and optimization

Reviewer: Similarly here, this looks like good content, but feels slightly out of place under headings about what quantum computers can do.

Quantum-inspired algorithms use quantum principles for increased speed and accuracy but implement on classical computer systems. This approach allows developers to leverage the power of new quantum techniques today without waiting for quantum hardware, which is still an emerging industry.

> Optimization is the task of balancing a system for maximum efficiency and minimum cost and is a crucial issue for industry and science. Quantum-inspired optimization algorithms running on today's classical computers can find solutions that up to now have not been possible. In addition to optimizing traffic flow to reduce congestion, there is airplane gate assignment, package delivery, job scheduling and more. With breakthroughs in materials science, there will be new forms of energy, batteries with larger capacity, and lighter and more durable materials.

Reviewer: Not all optimization tasks seek to increase efficiency of a process; while that is an important example, it's one example of what optimization problems can look like.

> [!NOTE]
> Read more about how Microsoft quantum-inspired computing is being used in [materials science](https://cloudblogs.microsoft.com/quantum/2020/01/21/oti-lumionics-accelerating-materials-design-microsoft-azure-quantum/), [risk management](https://cloudblogs.microsoft.com/quantum/2019/05/22/microsoft-quantum-collaborates-with-willis-towers-watson-to-transform-risk-management-solutions/), and [medicine](https://blogs.microsoft.com/blog/2018/05/18/microsoft-quantum-helps-case-western-reserve-university-advance-mri-research/).

## Quantum machine learning

Machine learning on classical computers is revolutionizing the world of science and business. However, the high computational cost of training the models hinders the development and scope of the field. The field of quantum machine learning explores how to devise and implement quantum software that could enable machine learning that is faster than that of classical computers.

> The Quantum Development Kit comes with the [quantum machine learning library](xref:microsoft.quantum.machine-learning.concepts.intro) that gives you the ability to run hybrid quantum/classical machine learning experiments. The library includes samples and tutorials, and provides the necessary tools to implement a new hybrid quantum-classical algorithm, the circuit-centric quantum classifier, to solve supervised classification problems.

Reviewer: ```suggestion
The Quantum Development Kit comes with the [quantum machine learning library](xref:microsoft.quantum.machine-learning.concepts.intro) that gives you the ability to run hybrid quantum/classical machine learning experiments. The library includes samples and tutorials, and provides the necessary tools to implement a new hybrid quantum–classical algorithm, the circuit-centric quantum classifier, to solve supervised classification problems.
```

## Q# and the Microsoft Quantum Development Kit (QDK)

Q# is Microsoft's open-source programming language for developing and running quantum algorithms. It is part of the [QDK](https://docs.microsoft.com/quantum/), a full-featured development kit for Q# that you can use with standard tools and languages to develop quantum applications that you can run in various environments, including the built-in full-state quantum simulator.

> There are extensions for Visual Studio and VS Code, and packages for use with Python and Jupyter Notebooks.

Reviewer: ```suggestion
There are extensions for Visual Studio and VS Code, and packages for use with Python and Jupyter Notebook.
```

> The QDK includes a standard library along with customized chemistry, machine learning, and numerics libraries.

Reviewer: I'd say specialized rather than customized here.

The documentation includes a Q# language guide, tutorials, and sample code to get you started quickly, and rich articles to help you dive deeper into quantum computing concepts.  

## Microsoft quantum hardware partners

> Microsoft is partnering with quantum hardware companies to provide developers with cloud access to quantum computers. Leveraging the [Azure Quantum](https://azure.microsoft.com/services/quantum/) platform and the Q# language, developers will be able to explore quantum algorithms and run their quantum programs on different types of quantum hardware.

Reviewer: ```suggestion
Microsoft is partnering with quantum hardware companies to provide developers with cloud access to quantum hardware. Leveraging the [Azure Quantum](https://azure.microsoft.com/services/quantum/) platform and the Q# language, developers will be able to explore quantum algorithms and run their quantum programs on different types of quantum hardware.
```
Alternatively, "quantum devices" could work.

+@KittyYeungQ for awareness.
